### ISSUE TITLE
Allow custom field types

### ISSUE THREAD

**Issue by fzaninotto** [marmelab/ng-admin#258]
Users should be able to add their custom field type. Currently they can handle this case with a `template` field type, but the app should allow to register a new directive under a certain field name.

```
app.registerField('datetime', app.module('myApp').directive(''myWriteDateTimeDirective', function() { //... }));
app.registerColumn('datetime', app.module('myApp').directive('myReadDateTimeDirective', function() { //... }));
```

And while we are at it, maybe our own directives should also be handled that way.

**Comment by Xennis** [comment]
That would be a nice feature.

**Comment by fzaninotto** [comment]
Fixed by #294
